test/rules/assertions/validTypes.js

```diff
@@ -180,12 +180,80 @@ export default {
       `,
       errors: [{
         line: 3,
-        message: 'Syntax error in type: ',
+        message: 'Tag @callback must have a namepath',
       }],
       options: [{
         allowEmptyNamepaths: false,
       }],
     },
+    {
+      code: `
+          /**
+           * @constant {str%ng}
+           */
+           const FOO = 'foo';
+      `,
+      errors: [
+        {
+          line: 3,
+          message: 'Syntax error in type: str%ng',
+        },
+      ],
+    },
+    {
+      code: `
+          /**
+           * @typedef {str%ng} UserString
+           */
+      `,
+      errors: [
+        {
+          line: 3,
+          message: 'Syntax error in type: str%ng',
+        },
+      ],
+    },
+    {
+      code: `
+          /**
+           * @typedef {string} UserStr%ng
+           */
+      `,
+      errors: [
+        {
+          line: 3,
+          message: 'Syntax error in namepath: UserStr%ng',
+        },
+      ],
+    },
+    {
+      code: `
+          /**
+           * @extends
+           */
+           class Bar {};
+      `,
+      errors: [
+        {
+          line: 3,
+          message: 'Tag @extends must have either a type or namepath',
```

Comment:
These error message changes are quite helpful, though in this particular case, I think it would be clearer to say `Tag @extends must have a type/namepath.` since it otherwise seems to suggest to me that these might be two separate components (like `@namespace {type} namepath`).

Reply:
Btw, I would think we should require `settings.jsdoc.mode` be set to `gcc` or `typescript` for  our existing test allowing `@extends {Foo<String>}` since the jsdoc spec doesn't indicate this format. In such a case, we can add one test which fails (with mode "jsdoc" or default) and one which passes (for "gcc" or "typescript").

Reply:
> Btw, I would think we should require `settings.jsdoc.mode` be set to `gcc` 

True, this is template class syntax of GCC and not part of vanilla JSDoc. I'm not certain on typescript since I don't really know ts dialect of JSDoc.